### DISCUSSION TITLE
Problem with ppCurrent font selection

### DISCUSSION THREAD

**Discussion by refaelsh** [xmonad/xmonad#512] [Q&A] [answered]
I have a problem with font selection in Xmobar. I've opened an issue here: https://codeberg.org/xmobar/xmobar/issues/713.
They suggested that it is probably an Xmonad issue.

What do you guys think?

**Comment by geekosaur** [comment, marked as answer]
I don't see how it could be our issue; as far as xmonad(-contrib) is concerned, they're just strings, and in particular we don't know anything about font settings. If it were only `xmobarFont` that were doing it, it might be our problem, but your original version can't be caused by xmonad, it's a literal string that is only interpreted by xmobar.

The addendum, however, is correct: if you use `<fn=1>` you're getting the first entry in `additionalFonts`, not the one in `font`. Remember that computers are almost always 0-based, not 1-based.

**Reply by refaelsh** [reply]
For future googlers: the issue no longer reproduces, I did not do anything.
I've probably misunderstood the entire situation.
Sorry to have been a bother :-)

**Comment by geekosaur** [comment]
Perhaps you needed to restart xmobar (or xmonad)? mod-q reliably restarts xmonad, but unreliably restarts xmobar because there's no reliable way to track child processes (and no way at all if it's started _alongside_ xmonad instead of _by_ it).

**Reply by refaelsh** [reply]
That was probably it. I was in the middle of tinkering with my config and probaby forgot to do it.
